### ISSUE TITLE
Input filed is not empty when selected is []

### ISSUE THREAD

**Issue by krembo** [ericgio/react-bootstrap-typeahead#87]
I like this component very much! However there is one issue that I've found.
When you are using the controlled comoponent and passing 'selected' parameter as an empty array,
the textfiled isn't cleared out. The previous selection is shown.

**Comment by ericgio** [comment]
Are you trying to clear out the component by passing in an empty array? Note that there is a [`clear`](https://github.com/ericgio/react-bootstrap-typeahead#clear) method to do that.

**Comment by krembo** [comment]
Yeah, I know. But if my form controlled by some object from store, I expect this form field to be updated when object resetted to initial - empty strings, arrays etc.
Like the other form fields. 
I don't get why I should handle this separately, if I'm already passing the value to the field.

**Comment by krembo** [comment]
I have to add that with multiple selection this part works as expected. Empty array - no selections in the field.
I'm talking about single selection. Probably should to clarify that.

**Comment by ericgio** [comment]
I'll take a look. If I recall, this came up previously and there were some complications since in your case, you want the component to be entirely cleared, but if I have a selection and hit backspace, that also clears the selected item but I want to maintain the text. It's difficult to tell these two scenarios apart and make the component behave according to each one.

**Comment by krembo** [comment]
That's what I thought also... Thanks for response anyway!
